src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -0,0 +1,107 @@
+        {
+            if (json == null)
+            {
+                throw new ArgumentNullException(nameof(json));
+            }
+
+            WriteRawValue(json.AsSpan(), skipInputValidation);
+        }
+
+        /// <summary>
+        /// Writes the input as JSON content.
+        /// </summary>
+        /// <param name="json">The raw JSON content to write.</param>
+        /// <param name="skipInputValidation">Whether to skip validation of the input JSON content.</param>
+        public void WriteRawValue(ReadOnlySpan<char> json, bool skipInputValidation = false)
+        {
+            byte[]? tempArray = null;
+
+            // For performance, avoid obtaining actual byte count unless memory usage is higher than the threshold.
+            Span<byte> utf8Json = json.Length <= (JsonConstants.ArrayPoolMaxSizeBeforeUsingNormalAlloc / JsonConstants.MaxExpansionFactorWhileTranscoding) ?
+                // Use a pooled alloc.
+                tempArray = ArrayPool<byte>.Shared.Rent(json.Length * JsonConstants.MaxExpansionFactorWhileTranscoding) :
+                // Use a normal alloc since the pool would create a normal alloc anyway based on the threshold (per current implementation)
+                // and by using a normal alloc we can avoid the Clear().
+                new byte[JsonReaderHelper.GetUtf8ByteCount(json)];
+
+            try
+            {
+                int actualByteCount = JsonReaderHelper.GetUtf8FromText(json, utf8Json);
+                utf8Json = utf8Json.Slice(0, actualByteCount);
+                WriteRawValue(utf8Json, skipInputValidation);
+            }
+            finally
+            {
+                if (tempArray != null)
+                {
+                    utf8Json.Clear();
+                    ArrayPool<byte>.Shared.Return(tempArray);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes the input as JSON content.
+        /// </summary>
+        /// <param name="utf8Json">The raw JSON content to write.</param>
+        /// <param name="skipInputValidation">Whether to skip validation of the input JSON content.</param>
+        public void WriteRawValue(ReadOnlySpan<byte> utf8Json, bool skipInputValidation = false)
+        {
+            if (utf8Json.Length == 0)
+            {
+                ThrowHelper.ThrowArgumentException(SR.ExpectedJsonTokens);
+            }
+
+            if (!skipInputValidation)
+            {
+                Utf8JsonReader reader = new Utf8JsonReader(utf8Json);
+
+                try
+                {
+                    while (reader.Read());
+                }
+                catch (JsonReaderException ex)
+                {
+                    ThrowHelper.ThrowArgumentException(ex.Message);
+                }
+            }
+
+            int maxRequired = utf8Json.Length + 1; // Optionally, 1 list separator
+
+            if (_memory.Length - BytesPending < maxRequired)
```

Comment:
Given the input json payload size could be quite large (more likely to be compared to a single JSON string token), it is a bit more important to try to copy in chunks rather than asking for a large contiguous buffer, similar to this issue:
https://github.com/dotnet/runtime/blob/ee0719079c4c4540f5adb135f002ad3ecf6c25ea/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.String.cs#L262

Might want to track that work, and follow up with user scenarios to see how important it is to copy in chunks.

Reply:
Thanks, this makes sense especially as these APIs have a higher chance of being called with large payloads. I added a comment with that issue as something to investigate as a follow-up.